### PR TITLE
Remove cached-property in favour of stdlib functools implementation

### PR DESCRIPTION
**Pull request by jpy-git**
### Brief summary of the change made

The `cached_property` is now available in the stdlib via `functools` for Python >= 3.8 so we should look to remove the `cached_property` package dependency.

See [here](https://github.com/pydanny/cached-property/issues/246)

I have used the backport for python 3.7, and we can phase this dependency out when 3.7 becomes end-of-life.

### Are there any other side effects of this change that we should be aware of?
No

### Pull Request checklist
- [X] Please confirm you have completed any of the necessary steps below.

- Included test cases to demonstrate any code changes, which may be one or more of the following:
  - `.yml` rule test cases in `test/fixtures/rules/std_rule_cases`.
  - `.sql`/`.yml` parser test cases in `test/fixtures/dialects` (note YML files can be auto generated with `tox -e generate-fixture-yml`).
  - Full autofix test cases in `test/fixtures/linter/autofix`.
  - Other.
- Added appropriate documentation for the change.
- Created GitHub issues for any relevant followup/future enhancements if appropriate.

### CODE REVIEW

File: plugins/sqlfluff-templater-dbt/sqlfluff_templater_dbt/templater.py [hunk 1]
```diff
@@ -4,12 +4,17 @@
...
 import os.path
 import logging
+import sys
 from typing import List, Optional, Iterator, Tuple, Any, Dict, Deque
 
 from dataclasses import dataclass
-from cached_property import cached_property
 from functools import partial
 
+if sys.version_info >= (3, 8):
```
Comment (barrywhart) [inline plugins/sqlfluff-templater-dbt/sqlfluff_templater_dbt/templater.py:13]: Nitpick: Do you think it'd be worth creating a module to encapsulate this code, e.g. `src/sqlfluff/core/cached_property.py`? That would save us having to repeat the version checking and conditional import logic in each file (current and future) that uses cached properties.

**Review by barrywhart — APPROVED** [review]
Nice cleanup! One nitpicky question about code reuse.

File: plugins/sqlfluff-templater-dbt/sqlfluff_templater_dbt/templater.py (hunk 1 above)
Comment (jpy-git) [reply plugins/sqlfluff-templater-dbt/sqlfluff_templater_dbt/templater.py:13]: yeah sure thing!

**Review by barrywhart — APPROVED** [review]
Great!

**Pull request merged**